pkgs/ts/@circles-pink/web-client/src/onboarding/utils/paginate.ts

```diff
@@ -0,0 +1,132 @@
+export const paginate = (
```

Comment:
There are a lot of calculations happening in this function. It has quite a few mutable variables and many boolean conditions. I'd feel more confident if we provide a test for it. An I think it would be a bit easier to understand with a type signature of the return type.